MockDataNaming.md

```diff
@@ -0,0 +1,101 @@
+# Mock Data Naming
+
+## Extension
+
+Mock data files should have an extension separate from any "extension" included from the path.  Here are the extensions that VOKMockUrlProtocol looks for, in the order in which they are searched:
+
+- `.http` files are parsed as full HTTP server responses, with
+    -  status on the first line
+    -  headers on the following line(s)
+    -  blank line
+    -  body on the following line(s)
+- `.json` files are parsed as JSON and served up with `HTTP/1.1 200 Success` and `Content-type: application/json`.
+
+## Base Name
+
+The base name consists of:
+
+- the HTTP method
+- a pipe character `|`
+- the request path
+- if there is a query string:
+	- a question mark `?`
+	- the query string (it is expected that the keys and values in a standard query string will be [percent-encoded](http://en.wikipedia.org/wiki/Percent-encoding) )
+- if there is a request body:
+	- a pipe character `|`
+	- the encoding of the request body, [as described below](#request-body)
+- any slashes `/` or colons `:` throughout are replaced by hyphens `-`
+
+### Request Body
+
+If the request's `Content-Type` header is `application/x-www-form-urlencoded`, use the content of the body.
+
+If the request's `Content-Type` header is one of (list subject to expansion):
+
+- `application/json`
+
+indicating a structured data format that can be decoded, then use the [bencoded](http://en.wikipedia.org/wiki/Bencode) form of the decoded data, represented as UTF-8 data, with [percent-encoding](http://en.wikipedia.org/wiki/Percent-encoding) applied to any non-unreserved characters.
```

Comment:
Nope, there's no reason whatsoever to expect these names to be any shorter.  I think the plan is still for Android to do the build-time translation of file names.